modules/core/04-channel/keeper/multihop_handshake_test.go

```diff
@@ -169,41 +170,122 @@ func (suite *MultihopTestSuite) TestChanOpenTryMultihop() {
+
+			suite.Z().Chain.CreatePortCapability(suite.Z().Chain.GetSimApp().ScopedIBCMockKeeper, ibctesting.MockPort)
+			portCap = suite.Z().Chain.GetPortCapability(ibctesting.MockPort)
+
+			heightDiff = 3 // consensus state doesn't exist at this height
+		}, false},
+		{"channel verification failed", func() {
+			// not creating a channel on chainA will result in an invalid proof of existence
+			suite.SetupConnections()
+			portCap = suite.Z().Chain.GetPortCapability(ibctesting.MockPort)
+		}, false},
+		{"port capability not found", func() {
+			suite.SetupConnections()
+			suite.chanPath.SetChannelOrdered()
+			err := suite.A().ChanOpenInit()
+			suite.Require().NoError(err)
+
+			portCap = capabilitytypes.NewCapability(3)
+		}, false},
+		{"connection version not negotiated", func() {
+			suite.SetupConnections()
+			suite.chanPath.SetChannelOrdered()
+			err := suite.A().ChanOpenInit()
+			suite.Require().NoError(err)
+
+			// modify A counterparty's versions
+			chain := suite.A().Endpoint.Counterparty
+			conn := chain.GetConnection()
+
+			version := connectiontypes.NewVersion("7", []string{"ORDER_ORDERED", "ORDER_UNORDERED"})
+			conn.Versions = append(conn.Versions, version)
+
+			chain.Chain.App.GetIBCKeeper().ConnectionKeeper.SetConnection(
+			    chain.Chain.GetContext(),
+			    chain.ConnectionID, conn,
+			)
+
+			suite.Z().Chain.CreatePortCapability(suite.Z().Chain.GetSimApp().ScopedIBCMockKeeper, ibctesting.MockPort)
+			portCap = suite.Z().Chain.GetPortCapability(ibctesting.MockPort)
+		}, false},
+		{"connection does not support ORDERED channels", func() {
+			suite.SetupConnections()
+			suite.chanPath.SetChannelOrdered()
+			err := suite.A().ChanOpenInit()
+			suite.Require().NoError(err)
+
+			// modify connA versions to only support UNORDERED channels
+			conn := suite.chanPath.EndpointA.GetConnection()
+
+			version := connectiontypes.NewVersion("1", []string{"ORDER_UNORDERED"})
+			conn.Versions = []*connectiontypes.Version{version}
+
+			suite.A().Chain.App.GetIBCKeeper().ConnectionKeeper.SetConnection(
+				suite.A().Chain.GetContext(),
+				suite.chanPath.EndpointA.ConnectionID, conn,
+			)
+			suite.A().Chain.CreatePortCapability(suite.A().Chain.GetSimApp().ScopedIBCMockKeeper, ibctesting.MockPort)
+			portCap = suite.A().Chain.GetPortCapability(ibctesting.MockPort)
+		}, false},
 	}
 
 	for _, tc := range testCases {
 		tc := tc
 		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
 			suite.SetupTest() // reset
-			tc.malleate()     // call ChanOpenInit and setup port capabilities
+			heightDiff = 0
+			tc.malleate() // call ChanOpenInit and setup port capabilities
+
+			if suite.chanPath.EndpointZ.ClientID != "" {
+				// update client on chainB
+				err := suite.chanPath.EndpointZ.UpdateClient()
+				suite.Require().NoError(err)
+			}
+
+			proof, err := suite.A().QueryChannelProof(suite.A().Chain.LastHeader.GetHeight())
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+			} else {
```

Comment:
linter is saying we can simplify this else/if condition check: `can replace 'else {if cond {}}' with 'else if cond {}'`

Reply:
Fixed